algorithms/nautilus/fuzzer/fuzzer.cpp

```diff
@@ -0,0 +1,278 @@
+/*
+ * fuzzuf
+ * Copyright (C) 2022 Ricerca Security
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+/**
+ * @file fuzzer.cpp
+ * @brief Fuzzing loop of Nautilus.
+ * @author Ricerca Security <fuzzuf-dev@ricsec.co.jp>
+ */
+#include <fstream>
+#include <memory>
+#include <nlohmann/json.hpp>
+#include <string>
+#include <vector>
+#include "fuzzuf/algorithms/nautilus/fuzzer/fuzzer.hpp"
+#include "fuzzuf/algorithms/nautilus/fuzzer/mutation_hierarflow_routines.hpp"
+#include "fuzzuf/algorithms/nautilus/fuzzer/other_hierarflow_routines.hpp"
+#include "fuzzuf/algorithms/nautilus/fuzzer/update_hierarflow_routines.hpp"
+#include "fuzzuf/hierarflow/hierarflow_intermediates.hpp"
+#include "fuzzuf/hierarflow/hierarflow_node.hpp"
+#include "fuzzuf/hierarflow/hierarflow_routine.hpp"
+#include "fuzzuf/logger/logger.hpp"
+#include "fuzzuf/utils/common.hpp"
+#include "fuzzuf/utils/filesystem.hpp"
+
+
+namespace fuzzuf::algorithm::nautilus::fuzzer {
+
+using json = nlohmann::json;
+
+/**
+ * @fn
+ * @brief Construct Nautilus fuzzer
+ * @param (state_ref) Reference to the state of Nautilus fuzzer
+ */
+NautilusFuzzer::NautilusFuzzer(std::unique_ptr<NautilusState>&& state_ref)
+  : state(std::move(state_ref)) {
+  /* Check files and load grammar */
+  CheckPathExistence();
+  LoadGrammar();
+
+  /* Construct fuzzing loop */
+  BuildFuzzFlow();
+
+  /* Clear screen */
+  MSG(TERM_CLEAR);
+}
+
+/**
+ * @fn
+ * @brief Build HierarFlow of Nautilus
+ */
+void NautilusFuzzer::BuildFuzzFlow() {
+  using fuzzuf::hierarflow::CreateNode;
+  using namespace fuzzuf::algorithm::nautilus::fuzzer::routine::other;
+  using namespace fuzzuf::algorithm::nautilus::fuzzer::routine::mutation;
+  using namespace fuzzuf::algorithm::nautilus::fuzzer::routine::update;
+
+  fuzz_loop = CreateNode<FuzzLoop>(*state);
+
+  /* Main flow */
+  auto select_input     = CreateNode<SelectInput>(*state);
+  auto process_input_or = CreateNode<ProcessInput>(*state);
```

Comment:
My understanding is that Nautilus takes two different ways to test a new input: mutating existing inputs or generating new ones. 
Probably you didn't use a word like "mutate" because ProcessInput has an unrelated phase of minimizing inputs.
While I also think this is decent, processing inputs is a bit vague. So I prefer some wording like "process_chosen_input_or" or "process_existing_input_or".

Reply:
Modified in [0ae818a](https://github.com/fuzzuf/fuzzuf/pull/23/commits/0ae818a3c23941c8450ec10884bb5966cfdf0419).